src/bdiff.c

```diff
@@ -69,6 +82,39 @@ static unsigned find_end_delta(
     return end_delta;
 }
 
+static unsigned slidey_aligner(
+        unsigned const sample_size, data_seeker const ds,
+        data_fetcher const df, char * const buf_fixed,
+        char * const buf_sliding, void * const fixed, void * const sliding,
+        unsigned const fixed_start, unsigned const sliding_end,
+        unsigned slide_distance) {
+    for (; slide_distance; slide_distance--) {
+        ds(sliding, sliding_end - slide_distance);
+        ds(fixed, fixed_start);
+        assert(df(fixed, buf_fixed, 1) == 1);
+        assert(df(sliding, buf_sliding, 1) == 1);
+        unsigned i = 0;
+        for (; i < sample_size; i++) {
+            if (buf_fixed[i] != buf_sliding[i]) {
+                break;
+            }
+        }
+        if (i != sample_size) {
+            continue;
+        }
+        unsigned const fsd = find_start_delta(
+            df, sample_size, buf_fixed, buf_sliding, fixed, sliding);
+        if (slide_distance == min(slide_distance, fsd)) {
```

Comment:
I could do with either some comments on these conditions, or an overview of the overall strategy in some docstrings!